### ISSUE TITLE
关于KG2E的score函数问题

### ISSUE THREAD

**Issue by Arseniolee** [DukeCheung/KG2E_Implementation#1]
感谢分享代码，另有两个问题是关于计算KL散度下的得分函数，望和您交流讨论：
def KLScore(self, **kwargs):
        # Calculate KL(e, r)
        losep1 = torch.sum(kwargs["errorv"]/kwargs["relationv"], dim=1)
        losep2 = torch.sum((kwargs["relationm"]-kwargs["errorm"])**2 / kwargs["relationv"], dim=1)
        # losep3 is added by Duke
        # losep3 = torch.sum(torch.log(kwargs["errorv"])-torch.log(kwargs["relationv"]), dim=1)
        KLer = (losep1 + losep2 - self.ke) / 2
        
        # Calculate KL(r, e)
        losep1 = torch.sum(kwargs["relationv"]/kwargs["errorv"], dim=1)
        losep2 = torch.sum((kwargs["errorm"] - kwargs["relationm"]) ** 2 / kwargs["errorv"], dim=1)
        KLre = (losep1 + losep2 - self.ke) / 2
        return (KLer + KLre) / 2

根据原文，losep1中应该有一个求矩阵的迹(tr)的过程，代码中并没有体现，同样，您增加了losep3，原文里是对矩阵的秩(det)求log，这里的代码也没有写出来。
另一个问题，原文提到过有一个基于KL散度的对称式得分函数，即(KLer + KLre) / 2，但同时指出这样的公式并没有提升性能，所以使用了非对称式的得分函数，也即只有KLer部分。

**Comment by DukeCheung** [comment]
不好意思，之前没注意，求矩阵的trace这个部分，我看了下，由于矩阵并非n×n的矩阵，所以未能实现。矩阵的秩当时也有加上，但是加上之后loss立马就变成nan了，后续也没有改。对称式的score函数，原文中确实有提到对性能并无帮助，这个可随意。
这个是我之前毕业论文时的内容，时间过得有点久了。如有不当处，万望包涵。
